doc/neps/nep-0028-website-redesign.rst

```diff
@@ -0,0 +1,302 @@
+   `Just The Docs <https://jekyllthemes.io/theme/just-the-docs>`_) that are
+   appropriate and free. Most themes are likely responsive for mobile, and
+   that should be a requirement. Jekyll uses a combination of liquid templating
+   and YAML to render HTML, and content is written in Markdown. i18n
+   functionality is not native to Jekyll, but can be added easily.
+2. *Hugo.* This is another well maintained option with 554 contributors, with
+   contributions within the last month. Hugo is written in Go, and similar to
+   Jekyll, has a simple to use CLI interface to generate static sites. Again,
+   similar to Jekyll, Hugo has a large directory of
+   `themes <https://themes.gohugo.io>`_. These themes appear to be free,
+   unlike some of Jekyll's themes.
+   (`Sample landing page theme <https://themes.gohugo.io/hugo-hero-theme>`_,
+   `docs theme <https://themes.gohugo.io/hugo-whisper-theme>`_). Hugo uses Jade
+   as its templating language, and content is also written in Markdown. i18n
+   functionality is native to Hugo.
+3. *Docusaurus.* Docusaurus is a responsive static site generator made by Facebook.
+   Unlike the previous options, Docusaurus doesn't come with themes, and thus we
+   would not want to use this for our landing page. This is an excellent docs
+   option written in React. Docusaurus natively has support for i18n (via
+   Crowdin_, document versioning, and document search.
+
+Both Jekyll and Hugo are excellent options that should be supported into the
+future and are good choices for NumPy. Docusaurus has several bonus features
+such as versioning and search that Jekyll and Hugo don't have, but is likely
+a poor candidate for a landing page - it could be a good option for a
+high-level docs site later on though.
+
+Deployment
+~~~~~~~~~~
+
+There is no need for running a server, and doing so is in our experience a
+significant drain on the time of maintainers.
+
+1. *Netlify.* Using netlify is free until 100GB of bandwidth is used. Additional
+   bandwidth costs $20/100GB. They support a global CDN system, which will keep
+   load times quick for users in other regions. Netlify also has Github integration,
+   which will allow for easy deployment. When a pull request is merged, Netlify
+   will automatically deploy the changes. DNS is simple, and HTTPS is also supported.
+2. *Github Pages.* Github Pages also has a 100GB bandwidth limit, and is unclear if
+   additional bandwidth can be purchased. It is also unclear where sites are deployed,
+   and should be assumed sites aren't deployed globally. Github Pages has an easy to
+   use CI & DNS, similar to to Netlify. HTTPS is supported.
+3. *Cloudflare.* An excellent option, additional CI is likely needed for the same
+   ease of deployment.
+
+All of the above options are appropriate for the NumPy site based on current
+traffic. Updating to a new deployment strategy, if needed, should take
+several hours at most and is inconsequential. If a provider such as
```

Comment:
Not clear if "Updating to a new deployment strategy ..." is a requirement or a way of stating that it is not viewed as an issue. If it truly is inconsequential than it should not be part of the NEP.

Reply:
> I don't understand the need/requirement for translation or i18n. None of the sites that we are considering as examples have it, and I have not seen any other sites that have it. Maybe I am missing something. Can you show an example of a website with translation?

[docusaurus](https://docusaurus.io/) provides the inbuilt feature for versioning and translation using Crowdin. So if the translation is not present then user can go to Crowdin site for numpy and translate the sentence one by one. 

> By the way, I know this is not the topic of this NEP, but I think that we should follow the good examples for the documentation of the code as well. For example Jupiter and Julia:
> 
> * https://jupyter.org/documentation
> * https://docs.julialang.org/en/v1/

Thanks for sharing it.

Reply:
Okay, I'll rephrase this.

Reply:
> First of all, choosing a few existing sites as examples to follow is the best thing to do.

This might save a little development time, but _only_ if we want to copy a good amount of content from these pages. Otherwise, I think most of the themes will give us a good starting point and we would avoid inheriting any 'hacks' the developers of those sites may have used.

Reply:
Good point @joelachance. I don't think there's much content to copy.

Reply:
I'll admit to writing this as well :) This should probably be removed. I would like to point out I think all the deployment options outlined in the NEP are all good and reasonable, and we wouldn't go wrong with any of them.